optuna/study/_optimize.py

```diff
@@ -189,6 +189,7 @@ def _run_trial(
         optuna.storages.fail_stale_trials(study)
 
     trial = study.ask()
+    study._thread_local.cache_all_trials.activate(lambda: study.get_trials(deepcopy=False))
```

Comment:
Is there an intention behind the design of making the argument to `activate` a lambda/configurable?

Reply:
not really. I will remove callable argument to simplify the code